include/suncontrol/suncontrol_i.h

```diff
@@ -0,0 +1,72 @@
+/* -----------------------------------------------------------------
+ * Programmer(s): Daniel R. Reynolds @ SMU
+ * -----------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2023, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------
+ * This is the header file for the SUNControl_I module.
+ * -----------------------------------------------------------------*/
+
+#ifndef _SUNCONTROL_I_H
+#define _SUNCONTROL_I_H
+
+#include <stdio.h>
+#include <sundials/sundials_control.h>
+
+#ifdef __cplusplus  /* wrapper to enable C++ usage */
+extern "C" {
+#endif
+
+/* ------------------------------
+ * I implementation of SUNControl
+ * ------------------------------ */
+
+struct _SUNControlContent_I {
+  realtype k1;        /* internal controller parameters */
+  realtype bias;      /* error bias factor */
+  int p;              /* order of accuracy to use for controller */
+  sunbooleantype pq;  /* p is embedding order (FALSE) or method order (TRUE) */
+};
+
+typedef struct _SUNControlContent_I *SUNControlContent_I;
+
+/* ------------------
+ * Exported Functions
+ * ------------------ */
+
+SUNDIALS_EXPORT
+SUNControl SUNControlI(SUNContext sunctx);
```

Comment:
It's not mentioned in the dev guide but we should probably follow the same naming convention for constructors as well i.e., `SUNControl_<implementation name>`, here and for the other controllers.